src/components/SettingComponent.js

```diff
@@ -148,17 +149,19 @@ export const PopoverColorPicker = ( { name, value, icon, storageKey, onChange, .
 
 PopoverColorPicker.propTypes = {
   value: string.isRequired,
-  icon: any.isRequired,
   name: string.isRequired,
   storageKey: string.isRequired,
   onChange: func.isRequired,
 }
 
+export const SkipRender = () => null
+
 const typeComponents = {
   [OPTION_TYPES.dropdown]: Dropdown,
   [OPTION_TYPES.toggle]: Toggle,
   [OPTION_TYPES.slider]: Slider,
   [OPTION_TYPES.popoverColorPicker]: PopoverColorPicker,
+  [OPTION_TYPES.skipRendering]: SkipRender,
```

Comment:
I meant this component, it seems a little wasteful. Instead use the absence of type` to skip the render/return null.